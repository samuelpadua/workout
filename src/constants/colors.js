export default {
  default: '#0086FF',
  success: '#00D604',
  warning: '#FF8A00',
  error: '#FF253A',
  lightGray: '#E5E5E5',
  gray: '#8C8C8C',
  mineShaft: '#404040'
}
