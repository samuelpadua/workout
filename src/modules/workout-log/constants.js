export const WORKOUT_LOG_ADD = 'workoutLog/ADD'
export const WORKOUT_LOG_REMOVE = 'workoutLog/REMOVE'
