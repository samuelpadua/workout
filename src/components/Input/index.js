import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Flex, Box } from '@rebass/grid'
import { colors } from '../../constants'

const InputStyled = styled.input`
  position: relative;
  border-radius: 4px;
  box-sizing : border-box;
  border: ${
  props => (props.error
    ? `2px solid ${colors.error}`
    : `1px solid ${colors.gray}`)
};
  font: inherit;
  color: currentColor;
  width: 100%;
  height: 3em;
  margin: 0;
  padding: ${
  props => (props.error
    ? '5px 11px 6px'
    : '6px 12px 7px')
};

  &:focus {
    border-color: red;
    border: 2px solid ${
  props => (props.error
    ? colors.error
    : colors.default)
};
    padding: 5px 11px 6px;
    box-shadow: none !important;
    outline: 0;
  }
`

const LabelStyled = styled.label`
  font-size: 13px;
  font-weight: 700;
  color: ${colors.mineShaft};
`

const LabelErrorStyled = styled.span`
  font-size: 13px;
  color: ${colors.error};
`

function Input(props) {
  const {
    label,
    error,
    errorMessage,
  } = props

  return (
    <Flex flexDirection="column">
      <Box mb="4px">
        <LabelStyled>
          { label }
        </LabelStyled>
      </Box>
      <Box>
        <InputStyled {...props} />
      </Box>
      {error && (
        <Box mt="4px">
          <LabelErrorStyled>{ errorMessage }</LabelErrorStyled>
        </Box>
      )}
    </Flex>
  )
}

Input.propTypes = {
  error: PropTypes.bool,
  errorMessage: PropTypes.string,
  label: PropTypes.string,
}

Input.defaultProps = {
  error: false,
  errorMessage: '',
  label: '',
}

export default Input
