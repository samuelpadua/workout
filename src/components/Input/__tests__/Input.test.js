
import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import Input from '../index'
import { colors } from '../../../constants'

describe('Input', () => {
  describe('default', () => {
    it('should render Input', () => {
      const tree = renderer
        .create(<Input />)
        .toJSON()
  
      expect(tree).toMatchSnapshot()
    })
  
    it('should change color border on focus', () => {
      const tree = renderer.create(<Input />).toJSON()
      expect(tree.children[1].children[0]).toHaveStyleRule('border', `2px solid ${colors.default}`, {
        modifier: ':focus',
      })
    })
  })

  describe('error', () => {
    it('should render Input with error state', () => {
      const tree = renderer
        .create(<Input error />)
        .toJSON()
  
      expect(tree).toMatchSnapshot()
    })

    it('should change color border on focus error state', () => {
      const tree = renderer.create(<Input error />).toJSON()
      expect(tree.children[1].children[0]).toHaveStyleRule('border', `2px solid ${colors.error}`, {
        modifier: ':focus',
      })
    })
  })
})
