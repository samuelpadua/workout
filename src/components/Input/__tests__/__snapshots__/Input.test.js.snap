// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input default should render Input 1`] = `
.c1 {
  box-sizing: border-box;
  margin-bottom: 4px;
}

.c3 {
  box-sizing: border-box;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  position: relative;
  border-radius: 4px;
  box-sizing: border-box;
  border: 1px solid #8C8C8C;
  font: inherit;
  color: currentColor;
  width: 100%;
  height: 3em;
  margin: 0;
  padding: 6px 12px 7px;
}

.c4:focus {
  border-color: red;
  border: 2px solid #0086FF;
  padding: 5px 11px 6px;
  box-shadow: none !important;
  outline: 0;
}

.c2 {
  font-size: 13px;
  font-weight: 700;
  color: #404040;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <label
      className="c2"
    >
      
    </label>
  </div>
  <div
    className="c3"
  >
    <input
      className="c4"
      label=""
    />
  </div>
</div>
`;

exports[`Input error should render Input with error state 1`] = `
.c1 {
  box-sizing: border-box;
  margin-bottom: 4px;
}

.c3 {
  box-sizing: border-box;
}

.c5 {
  box-sizing: border-box;
  margin-top: 4px;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  position: relative;
  border-radius: 4px;
  box-sizing: border-box;
  border: 2px solid #FF253A;
  font: inherit;
  color: currentColor;
  width: 100%;
  height: 3em;
  margin: 0;
  padding: 5px 11px 6px;
}

.c4:focus {
  border-color: red;
  border: 2px solid #FF253A;
  padding: 5px 11px 6px;
  box-shadow: none !important;
  outline: 0;
}

.c2 {
  font-size: 13px;
  font-weight: 700;
  color: #404040;
}

.c6 {
  font-size: 13px;
  color: #FF253A;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <label
      className="c2"
    >
      
    </label>
  </div>
  <div
    className="c3"
  >
    <input
      className="c4"
      label=""
    />
  </div>
  <div
    className="c5"
  >
    <span
      className="c6"
    >
      
    </span>
  </div>
</div>
`;
